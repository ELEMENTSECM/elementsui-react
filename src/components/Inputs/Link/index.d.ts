import * as React from 'react';

export interface LinkProps {
        /**
         * Link label
         */
        label?: string;
        /**
         * Link href
         */
        href?: string;
        /**
         * Link disabled
         */
        disabled?: boolean;
        /**
         * User-defined styling
         */
        getStyles?: (...args: any[])=>any;
    }

    declare const Link: React.SFC<LinkProps>;
export default Link;