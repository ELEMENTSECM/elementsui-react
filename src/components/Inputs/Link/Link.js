import * as React from 'react';
import PropTypes from 'prop-types';
import { Link as UIFabLink } from 'office-ui-fabric-react/lib/Link';

/**Link example */
function Link({ label, href, disabled }) {
	return (
		<UIFabLink href={href} disabled={disabled}>
			{label}
		</UIFabLink>
	);
}

Link.propTypes = {
	/**Link label */
	label: PropTypes.string,
	/**Link href */
	href: PropTypes.string,
	/**Link disabled */
	disabled: PropTypes.bool
};
export default Link;
