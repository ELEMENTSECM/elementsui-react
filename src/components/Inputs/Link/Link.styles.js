export const getStyles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Link',
			{
				// place your styles here
			},
			className
		]
	};
};
