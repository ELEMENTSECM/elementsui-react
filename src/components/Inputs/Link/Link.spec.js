import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

describe('Link', () => {
	test('displays `evry.com` as href', () => {
        const tree = renderer
        .create(<Link htmlId="defaultLink" href="evry.com" />)
        .toJSON();
		expect(tree).toMatchSnapshot();
	});
});