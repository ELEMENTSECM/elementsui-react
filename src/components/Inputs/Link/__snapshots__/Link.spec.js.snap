// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link displays \`evry.com\` as href 1`] = `
<a
  aria-disabled={undefined}
  className="ms-Link root-32"
  href="evry.com"
  onClick={[Function]}
  target={undefined}
/>
`;
